app/javascript/packs/fetch/requester.js

```diff
@@ -16,39 +16,68 @@ export async function destroy(route) {
 }
 
 export async function post(route, body) {
-  return doFetchRequest(route, 'POST', body);
+  return doFetchRequest(route, 'POST', JSON.stringify(body));
+}
+
+/**
+ * Sends a post request with a file rather than JSON as its contents.
+ *
+ * @param route     API route
+ * @param body      A FormData object with the request params
+ */
+export async function postFile(route, body) {
+  return doFetchRequest(route, 'POST', body, true);
 }
 
 export async function patch(route, body) {
-  return doFetchRequest(route, 'PATCH', body);
+  return doFetchRequest(route, 'PATCH', JSON.stringify(body));
+}
+
+/**
+ * Gets and downloads a file to the user's computer.
+ *
+ * @param route         URL of the file to download
+ * @returns {Promise}
+ */
+export async function downloadFile(route) {
+  let response = await doFetchRequest(route, 'GET', undefined, false, false);
+  let data = await response.blob();
+  return download(data);
```

Comment:
Where is `download`? Are you trying to use [`browser.downloads.download`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/downloads)? We haven't asked for user/browser permissions for this.

Reply:
Oops!! This was left over from when I tried using download.js instead of `a href` for downloading files